base_report_to_printer/models/report.py

```diff
@@ -42,7 +25,8 @@ def print_document(self, cr, uid, ids, report_name, html=None,
             raise exceptions.Warning(
                 _('No printer configured to print this report.')
             )
-        return printer.print_document(report, document, report.report_type)
+        return printer.with_context(context).print_document(
+            report, document, report.report_type)
```

Comment:
This method seems very strange to me as it mixes old and new API.

Can't we make this method in API 8.0 ?

Otherwise we could want to keep `printer_id = report_obj.behaviour(cr, uid, report.id, context=context)[report.id]`